tensorflow/contrib/repeat/kernels/repeat_op_cpu_impl.cc

```diff
@@ -0,0 +1,168 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/util/work_sharder.h"
+
+namespace tensorflow{
+
+template <typename T>
+void RepeatCPUImpl(const Tensor& input,
+                   const typename TTypes<int32>::ConstFlat& repeats_flat,
+                   int axis, Tensor* output) {
+  auto input_flat = input.flat<T>();
+  auto output_flat = output->flat<T>();
+    
+  // a batch is inner axes > axis
+  size_t batch_size = 1;
+  int32 dims = input.shape().dims();
+  for (int32 i = axis + 1; i < dims; ++i) {
+    batch_size *= input.shape().dim_size(i);
+  }
+  int64 num_batch = input_flat.size() / batch_size;
```

Comment:
Maybe you can use `Tensor::flat_inner_outer_dims` to simplify the computation of indices?